### DISCUSSION TITLE
默认配置下Apple一些服务的问题解决思路【供参考】

### DISCUSSION THREAD

**Discussion by TwoBall** [kkkgo/PaoPaoDNS#156] [经验分享]
问题一：AppleID-订阅无法打开的问题
我抓包看了下相关域名解析记录为：
`comm-cohort.ess.apple.com 【FakeIP】
comm-main.ess.apple.com【FakeIP】
gateway.icloud.com【FakeIP】
apps.mzstatic.com【正常IP】
xp.apple.com【正常IP】
speedysub.itunes.apple.com【FakeIP】`
问题二：Mac地图变TOMTOM的问题
这个谷歌一下就出来了，是`*.ls.apple.com`走了代理。
解决办法
https://raw.githubusercontent.com/ACL4SSR/ACL4SSR/master/Clash/Apple.list 都加入force_recurse_list.txt列表。
应该可以简洁一些，不过不是很清楚CNAUTO具体的分流列表，所以不好测试。

**Comment by kkkgo** [comment]
建议加入force_dnscrypt_list，force_recurse_list一般不适用（可能递归失败）。
apple的大多数是anycast的IP，除非是云上贵州，连通性可能并非所有服务可用。如果没特别指明（USE_MARK_DATA目前跳过了apple处理），现在是一般当IP库非境内IP处理，由于是anycast的IP，这在普通DNS服务下不会有什么问题，当然这个配合fakeip转发的话如果节点访问性不佳，会出现无法访问的情况，可以在fakeip网关或者DNS加入相关规则。
总结：建议加入force_dnscrypt_list。或者有后续一份比较精确的域名列表的话可以更好地改善USE_MARK_DATA。

**Reply by kkkgo** [reply]
有空可以优化一下，或者有更好的apple域名列表参考源可以留言。

**Reply by TwoBall** [reply]
https://ruleset.skk.moe/List/domainset/apple_cdn.conf 这个规则包含 Apple, Inc. 在中华人民共和国完成工信部 ICP 备案和公安网备、且在中华人民共和国境内提供 HTTP 服务的域名。因该可以作为USE_MARK_DATA的数据来源。

**Comment by TwoBall** [comment]
话说Apple的服务确实各种复杂。我在将上述服务域名加到force_dnscrypt_list后，依旧出现了问题一。![image](https://github.com/kkkgo/PaoPaoDNS/assets/9460044/b2d03cdf-087f-48ea-ac61-d7b2204ffcd6)
![image](https://github.com/kkkgo/PaoPaoDNS/assets/9460044/13c533bd-1ecb-4cab-80f3-7d84a938e3f4)

**Reply by TwoBall** [reply]
图一是force_dnscrypt_list，图二是force_recurse_list。xp.apple.com这个域名应该是有CDN的。![屏幕截图 2024-05-31 003204](https://github.com/kkkgo/PaoPaoDNS/assets/9460044/a257ff5d-8797-4baa-8584-5bf97cf77f55)

**Reply by kkkgo** [reply]
有cdn的本身不会被IP库转发到FAKEIP。所以应该不需要加入任何list。

**Reply by TwoBall** [reply]
问题一出现在speedysub.itunes.apple.com这个域名上，我将他单独加入force_dnscrypt_list，其他规则还原后可以正常访问。但是奇怪的是其他规则也加入force_dnscrypt_list。就不能正常访问（异常缓慢）。对比应该只有gateway.icloud.com这个域名产生了什么影响。经测试gateway.icloud.com有ipv6，我决定同时将他加进force_recurse_list。

**Reply by kkkgo** [reply]
所以需要准确的列表。如果你把模糊的列表全都塞进去force_dnscrypt_list确实可能得不到你预期的结果。像本身有CDN，没有被FAKEIP的就不需要特别加入处理。

**Comment by kkkgo** [comment]
USE_MARK_DATA应该加了一些域名，可以重启等待数据更新或者拉一下新镜像试试。

**Reply by TwoBall** [reply]
好的，我来测试看看
